handlers/compute_unix_test.go

```diff
@@ -311,3 +311,143 @@ func (s *ComputeTestSuite) TestPullImage(c *check.C) {
+	deleteContainer("/c861f990-4472-4fa1-960f-65171b544c28")
+
+	rawEvent = loadEvent("./test_events/instance_activate_sigtrap", c)
+	reply = testEvent(rawEvent, c)
+	container, ok = utils.GetFieldsIfExist(reply.Data, "instanceHostMap", "instance", "+data", "dockerContainer")
+	if !ok {
+		c.Fatal("No ID found")
+	}
+	dockerClient = docker.GetClient(docker.DefaultVersion)
+	inspect, err = dockerClient.ContainerInspect(context.Background(), container.(types.Container).ID)
+	if err != nil {
+		c.Fatal("Inspect Err")
+	}
+	c.Check(inspect.Config.Image, check.Equals, "happykiki/sigtrap:v0.1.0")
+
+	rawEventDe = loadEvent("./test_events/instance_deactivate_sigtrap", c)
+	event, _, fields = unmarshalEventAndInstanceFields(rawEvent, c)
+
+	fields["stopTimeout"] = 15
+	jsonTimeout, tOk = event["data"].(map[string]interface{})["processData"].(map[string]interface{})
+	if tOk {
+		jsonTimeout["timeout"] = 15
+	}
+	rawEvent = marshalEvent(event, c)
+
+	startTime = time.Now()
+	replyDe = testEvent(rawEventDe, c)
+	container, ok = utils.GetFieldsIfExist(replyDe.Data, "instanceHostMap", "instance", "+data", "dockerContainer")
+	if !ok {
+		c.Fatal("No ID found")
+	}
+	inspect, err = dockerClient.ContainerInspect(context.Background(), container.(types.Container).ID)
+	if err != nil {
+		c.Fatal("Inspect Err")
+	}
+
+	endTime, err = time.Parse(time.RFC3339, inspect.State.FinishedAt)
+	if err != nil {
+		fmt.Println(err)
+	}
+	duration = endTime.Sub(startTime).Seconds()
+	if duration < 14 && duration > 16 {
+		c.Fatal("wrong container timeout value")
+	}
+
+	// case 3: stopTimeout < timeout
+	deleteContainer("/c861f990-4472-4fa1-960f-65171b544c28")
+
+	rawEvent = loadEvent("./test_events/instance_activate_sigtrap", c)
+	reply = testEvent(rawEvent, c)
+	container, ok = utils.GetFieldsIfExist(reply.Data, "instanceHostMap", "instance", "+data", "dockerContainer")
+	if !ok {
+		c.Fatal("No ID found")
+	}
+	dockerClient = docker.GetClient(docker.DefaultVersion)
+	inspect, err = dockerClient.ContainerInspect(context.Background(), container.(types.Container).ID)
+	if err != nil {
+		c.Fatal("Inspect Err")
+	}
+	c.Check(inspect.Config.Image, check.Equals, "happykiki/sigtrap:v0.1.0")
+
+	rawEventDe = loadEvent("./test_events/instance_deactivate_sigtrap", c)
+	event, _, fields = unmarshalEventAndInstanceFields(rawEvent, c)
+
+	fields["stopTimeout"] = 15
+	jsonTimeout, tOk = event["data"].(map[string]interface{})["processData"].(map[string]interface{})
+	if tOk {
+		jsonTimeout["timeout"] = 20
+	}
+	rawEvent = marshalEvent(event, c)
+
+	startTime = time.Now()
+	replyDe = testEvent(rawEventDe, c)
+	container, ok = utils.GetFieldsIfExist(replyDe.Data, "instanceHostMap", "instance", "+data", "dockerContainer")
+	if !ok {
+		c.Fatal("No ID found")
+	}
+	inspect, err = dockerClient.ContainerInspect(context.Background(), container.(types.Container).ID)
+	if err != nil {
+		c.Fatal("Inspect Err")
+	}
+
+	endTime, err = time.Parse(time.RFC3339, inspect.State.FinishedAt)
```

Comment:
i think you need to wait for container to be finished. That is to say, wait for `inspect.State.FinishedAt` to become available.